front/components/data_source_view/DataSourceViewSelector.tsx

```diff
@@ -29,30 +36,147 @@ import {
   isManaged,
   isWebsite,
 } from "@app/lib/data_sources";
+import { useVaults } from "@app/lib/swr/vaults";
+import { classNames } from "@app/lib/utils";
+import { getVaultIcon, getVaultName } from "@app/lib/vaults";
 
 const MIN_TOTAL_DATA_SOURCES_TO_GROUP = 12;
 const MIN_DATA_SOURCES_PER_KIND_TO_GROUP = 3;
+const ONLY_ONE_VAULT_PER_SELECTION = true;
 
 interface DataSourceViewsSelectorProps {
   owner: LightWorkspaceType;
   dataSourceViews: DataSourceViewType[];
+  allowedVaults?: VaultType[];
   selectionConfigurations: DataSourceViewSelectionConfigurations;
   setSelectionConfigurations: Dispatch<
     SetStateAction<DataSourceViewSelectionConfigurations>
   >;
   viewType: ContentNodesViewType;
 }
 
+function VaultSelector({
+  owner,
+  dataSourceViews,
+  allowedVaults,
+  selectionConfigurations,
+  setSelectionConfigurations,
+  viewType,
+}: DataSourceViewsSelectorProps) {
+  const dataSourceViewsByVaultId = useMemo(
+    () => _.groupBy(dataSourceViews, (dsv) => dsv.vaultId),
+    [dataSourceViews]
+  );
+
+  const { vaults, isVaultsError, isVaultsLoading } = useVaults({
+    workspaceId: owner.sId,
+  });
+
+  const [selectedVault, setSelectedVault] = useState<string>(
+    Object.keys(selectionConfigurations).length > 0
+      ? selectionConfigurations[Object.keys(selectionConfigurations)[0]]
+          .dataSourceView.vaultId
+      : ""
+  );
+
+  if (isVaultsLoading || isVaultsError || !vaults) {
+    return <Spinner />;
+  }
+
+  if (Object.keys(dataSourceViewsByVaultId).length === 1) {
```

Comment:
Is this to serve the case where the workspace does not have any vault (outside of the global one)?

Reply:
yes

Reply:
Let's add a small comment for the next one that wonders 😉

Reply:
I did (but forgot to push :p)

Reply:
Still don't see it 😛